### PR TITLE
Waves: add accessor for waves

### PR DESCRIPTION
**Pull request by gauteh**
For plotting elevation energy spectra. 

* This may have to be a dataset accessor because otherwise we do not have access to the time coordinate.
* In the future this may go into a separate package. But as long as we use the same accessor name the API should be the same for users.

```
    csv_in = test_data / 'csv/omb3.csv'
    ds = read_omb_csv(csv_in)
    print(ds)
    print(ds.elevation_energy_spectrum)
    print(ds.frequencies_waves_imu)

    plt.figure()
    ds.isel(trajectory=0).elevation_energy_spectrum.wave.plot(
        ds.isel(trajectory=0).time_waves_imu.squeeze())

    if plot:
        plt.show()

    plt.close('all')

```

Solves #134

### CODE REVIEW

File: trajan/waves/plot.py [hunk 1]
```diff
@@ -69,8 +52,10 @@ def spectra(self, time, *args, **kwargs):
         """
         vrange = kwargs.pop('vrange', None)
         nseconds_gap = kwargs.pop('nseconds_gap', 6 * 3600)
-
-        ax = self.set_up_map(**kwargs)
+        try:
+            ax = kwargs.pop('ax')
+        except:
+            ax = plt.axes()
```
Comment (gauteh) [inline trajan/waves/plot.py:58]: This catches all exceptions. It is better to act as predetermined as possible, if there are exceptions there is probably an issue we should fix.

**Comment by gauteh** [comment]
This is the way to do it!

File: trajan/waves/plot.py (hunk 1 above)
Comment (gauteh) [reply trajan/waves/plot.py:58]: Yes, they may be eagerly evaluated, maybe something like:

```
ax = kwargs.pop('ax', None)
ax = ax if ax is not None else plt.axes()
```

File: examples/example_plot_spectra_accessors.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+"""
+Plotting wave spectra data, using the accessor syntax
+================================================
```
Comment (gauteh) [inline examples/example_plot_spectra_accessors.py:3]: The documentation is going to fail here, for some reason the `===` need to be equal or longer than the title.

File: trajan/waves/plot.py (hunk 1 above)
Comment (gauteh) [reply trajan/waves/plot.py:63]: I think your solution looks good.

**Comment by gauteh** [comment]
Yep, you remove the old one. I think it is ready to merge. In the future we may have to move it up to dataset level, alternatively use to_1d after selecting trajectory and then plotting (then the times variable won't be necessary).

**Comment by knutfrode** [comment]
Good, then I merge now.

**Pull request merged**